the-collab-lab/tcl-44-smart-shopping-list
# Order shopping list items

## Description

- This PR orders the items by the timeframe (alphabetically if they are the same) and active/inactive status.
- Added `isActive` field in the `useAddItem` hook, which will be true when a user adds a new item
- We added `orderBy` from Firebase in `useFetchItems` hook, to sort the items by their active status first, then `timeframe` and `itemName`
- We used `isActive` in a useEffect to check if the item is inactive and update the docs.
- Created `getItemCategory` function to colour the checkboxes depending on their timeframe.

### Refactoring

- Created `dateHelpers` in utils to store things related to date that are used in multiple places.

## Related Issue

Closes #12 

## Acceptance Criteria

- Items in the list are shown as visually distinct (e.g., with a different background color on the list item) according to how soon the item is expected to be bought again: Soon, Kind of soon, Not soon, Inactive
- Items should be sorted by the estimated number of days until the next purchase
- Items with the same number of estimated days until next purchase should be sorted alphabetically
- Items in the different states should be described distinctly when read by a screen reader - ARIA label

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

![shows the list being ordered by timeframe and active status](https://user-images.githubusercontent.com/29425781/167931243-3d8299ef-2cfe-4bba-a484-06da77711b92.gif)

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with git pull origin rb-dd-order-shopping-list-items and check that branch out with git checkout rb-dd-order-shopping-list-items.
- Then npm start to launch the app.
- Add items to the shopping list.
- Change the timeframe in Firestore to either fewer than 7, between 7-30, or more than 30, you will notice the colours of the checkboxes will change.
- To test inactive state, set an item to 2 days timeframe and then change the lastPurchased date to 5 days (more than 2 times the timeframe) in the past. The item should move to the bottom of the list and be greyed out.
